packages/next-auth/src/providers/pinterest.ts

```diff
@@ -0,0 +1,34 @@
+import { OAuthConfig, OAuthUserConfig } from "."
+
+export interface PinterestProfile extends Record<string, any> {
+  account_type: "BUSINESS" | "PINNER"
+  profile_image: string
+  website_url: string
+  username: string
+}
+
+export default function PinterestProvider<P extends PinterestProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "pinterest",
+    name: "Pinterest",
+    type: "oauth",
+    authorization: {
+      url: "https://www.pinterest.com/oauth",
+      params: { scope: "user_accounts:read" },
+    },
+    checks: ["state"],
+    token: "https://api.pinterest.com/v5/oauth/token",
+    userinfo: "https://api.pinterest.com/v5/user_account",
+    profile({ username, profile_image }) {
+      return {
+        id: username,
+        name: username,
+        image: profile_image,
+        email: null,
```

Comment:
We don't want to expose more than these by default. Do you know if the e-mail can be retrieved from somewhere? Otherwise, this provider won't work with our database strategy.

Reply:
No, Pinterest doesn't provide email 
See docs: https://developers.pinterest.com/docs/api/v5/#tag/user_account

Reply:
I noticed a lot of existing providers such as Instagram do not provide user email as well